src/OrchardCore/OrchardCore.Setup.Abstractions/Events/ISetupEventHandler.cs

```diff
@@ -1,17 +1,35 @@
 using System;
 using System.Collections.Generic;
 using System.Threading.Tasks;
+using OrchardCore.Setup.Services;
 
-namespace OrchardCore.Setup.Events
+namespace OrchardCore.Setup.Events;
+
+public interface ISetupEventHandler
 {
+    [Obsolete($"This method is obsolete and will be removed in future releases. Please use '{nameof(SetupAsync)}' instead.", false)]
+    Task Setup(IDictionary<string, object> properties, Action<string, string> reportError)
+        => Task.CompletedTask;
+
+    /// <summary>
+    /// Called during the process of setting up a new tenant.
+    /// </summary>
+    /// <param name="context"></param>
+    /// <returns></returns>
+    Task SetupAsync(SetupContext context)
+        => Task.CompletedTask;
```

Comment:
In the meantime I will approve

But as a last idea the default implementation of `SetupAsync()` could be to call `Setup()` so that we don't need to call the obsolete `Setup()` from the `SetupService` and only use the rule pragma here, all in one place.

Something like this, quickly written so may need to be tweaked.

    Task SetupAsync(SetupContext context) =>
        Setup(context.Properties, (key, message) => context.Errors[key] = message );